packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx

```diff
@@ -36,8 +36,9 @@ export function UploadPreviewer({
   }`;
 
   const isDisabled =
-    fileStatuses.some((status) =>
-      ['error', 'editing'].includes(status?.fileState)
+    fileStatuses.some(
+      (status) =>
+        status?.fileState && ['error', 'editing'].includes(status?.fileState)
```

Comment:
Would it make sense to make `fileState` a required key of `status` rather than add this check?

Reply:
`fileState` a required key of `status`. Added this check as `null` is a valid value for `fileState`.

https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/components/Storage/FileUploader/types.ts#L80

Reply:
IMO we should address the usage of `null` as a possible value of `fileState` by disallowing it (since `null` is not really a state). Granted that we will have a larger impact overall, but it will improve the overall code and feature quality

Reply:
Talked to Eric and it looks like adding another state, maybe like `idle` will be a bigger refactor so will pull this change out of this PR and have a separate PR for `FileUploader` changes.